gadget/gadget.go

```diff
@@ -445,16 +445,59 @@ func readInfo(f func(string) ([]byte, error), gadgetYamlFn string, model Model)
 
 // ReadInfo reads the gadget specific metadata from meta/gadget.yaml in the snap
 // root directory.
+// See ReadInfoAndValidate for a variant that does consistency rules
+// validation like Validate.
 func ReadInfo(gadgetSnapRootDir string, model Model) (*Info, error) {
 	gadgetYamlFn := filepath.Join(gadgetSnapRootDir, "meta", "gadget.yaml")
-	return readInfo(ioutil.ReadFile, gadgetYamlFn, model)
+	ginfo, err := readInfo(ioutil.ReadFile, gadgetYamlFn, model)
+	if err != nil {
+		return nil, err
+	}
+	return ginfo, nil
+
+}
+
+// ReadInfoAndValidate reads the gadget specific metadata from
+// meta/gadget.yaml in the snap root directory.
+// It also performs consistency rules validation as Validate does
+// using the given constraints. See ReadInfo for a variant that does not.
```

Comment:
maybe also mention that some callers might also want to use ValidateContent on top of the validation that is performed here too, like we do in install mode devicestate handler?

Reply:
I'm adding:

```
// See also ValidateContent for further validating the content itself
// instead of the metadata.
```